### PR TITLE
[JENKINS-30820] Only regenerate StackMap for JDK6+ classes

### PR DESCRIPTION
**Pull request by jtnord**
This applies a sticky plaster onto the bct so that it does not generate stackMaps for JDK5 and lower bytecode.
The root cause still exists and can be hit for JDK6+ classes but this at least reduces the number of plugins affected.

This also adds a bucket load of logging and moves the test code into a package.

Tested with a locally patched jenkins and the batch-task-plugin

@reviewbybees /cc: @olivergondza 
[JENKINS-30820](https://issues.jenkins-ci.org/browse/JENKINS-30820)

### CODE REVIEW

File: src/main/java/org/jenkinsci/bytecode/AdaptField.java [hunk 1]
```diff
@@ -62,10 +62,12 @@ void parse(TransformationSpec spec, AnnotatedElement e) {
             if (name.length()==0)   name = mem.getName(); // default to the same name
 
             MemberAdapter mrs = null;
-            if (e instanceof Field)
+            if (e instanceof Field) {
                 mrs = fieldToField((Field) e);
-            if (e instanceof Method)
+            }
+            if (e instanceof Method) {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/bytecode/AdaptField.java]: unrelated changes :ant:

File: src/main/java/org/jenkinsci/bytecode/Transformer.java [hunk 2]
```diff
@@ -66,49 +78,83 @@ public synchronized void loadRules(Collection<? extends ClassLoader> loaders) th
...
         return cw.toByteArray();
     }
+    
+    /**
+     * Inspects a byte array representation of a class and returns the version of the bytecode.
+     * @param classData the class bytecode.
+     * @return an integer representing the major version of the bytecode.
+     */
+    private int getBytecodeVersion(byte[] classData) {
+        int version = (( classData[6] & 0xFF ) << 8 ) | (classData[7] & 0xFF);
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/bytecode/Transformer.java]: Is there any need of such masking in Java?

Comment (jtnord) [inline src/main/java/org/jenkinsci/bytecode/Transformer.java]: yes - its a byte so if the MSB is set this will be a negative number (0xff -> -1 -> 0xffffff when promoted to an int)

File: src/main/java/org/jenkinsci/bytecode/AdaptField.java (hunk 1 above)
Comment (jtnord) [inline src/main/java/org/jenkinsci/bytecode/AdaptField.java]: yes but it helped when debugging and we do not need to worry about merge conflicts and back-porting here as this is a linear release train (yay for almost CD)

**Comment by stephenc** [comment]
:bee:

**Comment by jtnord** [comment]
@reviewbybees done (but there will be a merge conflict with this and #6)
